blocks/media-upload-button/index.js

```diff
@@ -59,14 +59,14 @@ const slimImageObject = ( img ) => {
 };
 
 class MediaUploadButton extends Component {
-	constructor( { multiple = false, type, gallery = false } ) {
+	constructor( { multiple = false, type, gallery = false, title = __( 'Select or Upload a media' ) } ) {
```

Comment:
This is still poor grammar, even if not used in the specific case of featured image assignment. Maybe "Select or Upload Media".